packages/transformer/src/IModelTransformer.ts

```diff
@@ -1049,11 +1051,18 @@ export class IModelTransformer extends IModelExportHandler {
         );
       }
       if (!this._options.noProvenance) {
-        this.provenanceDb.elements.insertAspect({
+        const id = this.provenanceDb.elements.insertAspect({
           ...aspectProps,
           jsonProperties: JSON.stringify(aspectProps.jsonProperties) as any,
         });
+        aspectProps.id = id;
       }
+    } else {
+      aspectProps.id = foundEsaProps.aspectId;
+      aspectProps.version = foundEsaProps.version;
+      aspectProps.jsonProperties = foundEsaProps.jsonProperties
+        ? JSON.parse(foundEsaProps.jsonProperties)
+        : {};
```

Comment:
Yeah I think this might be left over and realized how weird this is when I was working on my other PR about handling imodels last synced before the version change. I will fix it up in my next PR if thats alright with you, because in order to support old imodels we probably need to instead setup the pendingreversesyncversion and pendingsyncversion jsonProperties if jsonProperties end up being undefined.